rtlib/shape.hh

```diff
@@ -51,6 +54,38 @@
 template <typename T, typename Size, typename alphset = ShapeAlph<T, Size> >
 class Fiber {
  private:
+#ifdef CHECKPOINTING_INTEGRATED
+    Size array_size;  // length/number of elements of array
```

Comment:
why does that variable need to be a member of the class? If I read it correct, you store it into the archive to know how many entries will follow, but is the value ever updated by the remainder of the program? You are calculating it once in `save`. Or am I missing something?

Reply:
Good point. It actually doesn't need to be a class member since I only need it as a temp variable in the `load` method, so I moved it into the `load` method as a local variable.